frontend/src/components/search/SearchOpportunityStatus.tsx

```diff
@@ -1,49 +1,88 @@
+];
+
+// Wait a half-second before updating query params
+// and submitting the form
+const SEARCH_OPPORTUNITY_STATUS_DEBOUNCE_TIME = 500;
+
+const SearchOpportunityStatus: React.FC<SearchOpportunityStatusProps> = ({
+  formRef,
+}) => {
+  const [mounted, setMounted] = useState(false);
+  const { updateQueryParams } = useSearchParamUpdater();
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [selectedStatuses, setSelectedStatuses] = useState<Set<string>>(
+    new Set(),
+  );
+
+  const debouncedUpdate = useDebouncedCallback(
+    (selectedStatuses: Set<string>) => {
+      const key = "status";
+      updateQueryParams(selectedStatuses, key);
+      formRef?.current?.requestSubmit();
+    },
+    SEARCH_OPPORTUNITY_STATUS_DEBOUNCE_TIME,
+  );
+
+  const handleCheck = (statusValue: string, isChecked: boolean) => {
+    setSelectedStatuses((prevSelectedStatuses) => {
+      const updatedStatuses = new Set(prevSelectedStatuses);
+      isChecked
+        ? updatedStatuses.add(statusValue)
+        : updatedStatuses.delete(statusValue);
+
+      debouncedUpdate(updatedStatuses);
+      return updatedStatuses;
+    });
+  };
+
+  useEffect(() => {
+    setMounted(true);
+    return () => {
+      setMounted(false);
+    };
+  }, []);
 
-export default function SearchOpportunityStatus() {
   return (
     <>
       <h4 className="margin-bottom-1">Opportunity status</h4>
-
       <div className="grid-row flex-wrap">
-        <div className="grid-col-6 padding-right-1">
-          <Checkbox
-            id="status-forecasted"
-            name="status-forecasted"
-            label="Forecasted"
-            tile={true}
-            className=""
-          />
-        </div>
-        <div className="grid-col-6 padding-right-1">
-          <Checkbox
-            id="status-posted"
-            name="status-posted"
-            label="Posted"
-            tile={true}
-            className=""
-          />
-        </div>
-        <div className="grid-col-6 padding-right-1">
-          <Checkbox
-            id="status-closed"
-            name="status-closed"
-            label="Closed"
-            tile={true}
-            className=""
-          />
-        </div>
-        <div className="grid-col-6 padding-right-1">
-          <Checkbox
-            id="status-archived"
-            name="status-archived"
-            label="Archived"
-            tile={true}
-            className=""
-          />
-        </div>
+        {statusOptions.map((option) => (
+          <div key={option.id} className="grid-col-6 padding-right-1">
+            <Checkbox
+              id={option.id}
+              name={option.id}
+              label={option.label}
+              tile={true}
+              onChange={(e) => handleCheck(option.value, e.target.checked)}
+              disabled={!mounted} // Required to be disabled until hydrated so query params are updated properly
```

Comment:
@andycochran @crystabelrangel - please note that I had to disable it temporarily until the component is loaded - it results in a small second where it is slightly grayer. 

The actual shell html is loaded before things become interactive, and this particularly effects the checkboxes